packages/collaboration/src/sharedlink.ts

```diff
@@ -45,22 +45,458 @@ export async function showSharedLinkDialog({
+      headers: {
+        Authorization: `token ${token}`
+      }
+    });
+    const data = await response.json();
+    const hubVersion = data.version;
+    const [major] = hubVersion.split('.').map(Number);
+    if (major >= 5) {
+      // The Hub version is compatible with the share feature, but we need to check if the user has rights to create a share
+      const userResponse = await fetch(`${hubApiUrl}/user`, {
+        headers: {
+          Authorization: `token ${token}`
+        }
+      });
+      const userData = await userResponse.json();
+      // The permissions needed are "read:users:name" (to be able to get a user by his/her name) and "shares!user=owner" (to be able to manage shares)
+      if (userData.scopes.includes('read:users:name')) {
+        // Check for shares permission (shares!user or shares!user=owner or shares!server=owner/serverName)
+        if (
+          (serverOwner === PageConfig.getOption('hubUser') &&
+            userData.scopes.includes('shares!user')) ||
+          userData.scopes.includes('shares!user=' + serverOwner) ||
+          userData.scopes.includes(
+            'shares!server=' + serverOwner + '/' + serverName
+          )
+        ) {
+          // If the user has the required permissions, we can create a share
+          canCreateShare = true;
+          // Check if the user has the correct scope to list all users (not mandatory, but makes the UI easier to use)
+          if (
+            userData.scopes.includes('list:users') ||
+            userData.scopes.includes('read:users') ||
+            userData.scopes.includes('admin:users')
+          ) {
+            canListUsers = true;
+          }
+          // Check if the user has the correct scope to list all groups (not mandatory, but makes the UI easier to use)
+          if (
+            userData.scopes.includes('list:groups') ||
+            userData.scopes.includes('read:groups') ||
+            userData.scopes.includes('admin:groups')
+          ) {
+            canListGroups = true;
+          }
+          // Check if the user has the correct scope to control the server (not mandatory, but allows other users to start/stop the server)
+          if (
+            (serverOwner === PageConfig.getOption('hubUser') &&
+              userData.scopes.includes('servers!user')) ||
+            userData.scopes.includes('servers!user=' + serverOwner) ||
+            userData.scopes.includes(
+              'servers!server=' + serverOwner + '/' + serverName + '/'
+            ) ||
+            userData.scopes.includes('admin:servers')
+          ) {
+            canControlServer = true;
+          }
+        }
+      }
+    }
+  }
+
+  // If we can create a share, open the proper UI
+  if (canCreateShare) {
+    let readableServerName = serverName;
+    if (readableServerName === '') {
+      readableServerName = 'default';
+    }
+    return showDialog({
+      title: trans.__('Share Jupyter Server ') + readableServerName,
```

Comment:
Different languages have different order of words

```suggestion
      title: trans.__('Share Jupyter Server %1', readableServerName),
```